packages/esm-care-panel-app/src/vl-history/vl-history.resource.ts

```diff
@@ -3,59 +3,38 @@ import { useMemo } from 'react';
 import useSWR from 'swr';
 import { configSchema } from '../config-schema';
 
-export function usePatientObservations(patientUuid, conceptUuids) {
-  const conceptsQueryParam = conceptUuids.join('%2C');
-  const apiUrl = `${fhirBaseUrl}/Observation?patient=${patientUuid}&code=${conceptsQueryParam}&_summary=data&_sort=-date&_count=12`;
+export function useViralLoadOservations(patientUuid) {
+  const apiUrl = `${fhirBaseUrl}/Observation?patient=${patientUuid}&code=${configSchema.viralLoadSetUuid._default}&_summary=data&_sort=-date`;
 
   const { data, error, isValidating, mutate } = useSWR<{ data: any }, Error>(apiUrl, openmrsFetch);
 
-  const observationsByDate = useMemo(() => {
-    if (!data) return {};
+  const observations = data?.data?.entry?.map((entry) => entry.resource);
 
-    const observations = data?.data?.entry?.map((entry) => entry.resource);
-
-    // Grouped observations by effectiveDateTime
-    const groupedObservations = observations?.reduce((acc, observation) => {
-      const { effectiveDateTime } = observation;
-      if (!acc[effectiveDateTime]) {
-        acc[effectiveDateTime] = {
-          observations: [],
-          valuesArray: [],
-          dateArray: [],
-          displayArray: [],
-        };
-      }
-      acc[effectiveDateTime]?.observations.push(observation);
-
-      const coding = observation?.code?.coding[0];
-      if (coding?.code === `${configSchema.hivViralLoadUuid._default}` && observation?.valueQuantity) {
-        acc[effectiveDateTime].valuesArray.push(observation.valueQuantity.value);
-      }
-
-      if (coding?.code === `${configSchema.hivViralLoadDateUuid._default}`) {
-        acc[effectiveDateTime].dateArray.push(observation.effectiveDateTime);
-      }
-
-      if (
-        coding?.code === `${configSchema.hivViralLoadQualitativeUuid._default}` &&
-        observation?.valueCodeableConcept
-      ) {
-        acc[effectiveDateTime].displayArray.push(observation.valueCodeableConcept?.coding[0]?.display);
-      }
-
-      if (acc[effectiveDateTime].dateArray.length === 0) {
-        acc[effectiveDateTime].dateArray.push(observation.effectiveDateTime);
-      }
-
-      return acc;
-    }, {});
-
-    return groupedObservations;
-  }, [data]);
   return {
-    observations: observationsByDate,
+    data: observations,
     isLoading: isValidating,
     isError: !!error,
     mutate,
   };
 }
+
+export function parseValueString(valueString) {
+  const parts = valueString.split(',').map((part) => part.trim());
+  let date, qualitative, viralLoad;
+
+  parts.forEach((part) => {
+    if (part.match(/^\d{4}-\d{2}-\d{2}$/)) {
+      date = part;
+    } else if (part === 'Target Not Detected' || part === 'DETECTED' || part === 'POOR SAMPLE QUALITY') {
```

Comment:
@Daphne210 What if Its another value that is introduced in the future what would happen.

Reply:
Then we adjust accordingly just like queue statuses of COMPLETED, PENDING, ...